### PR TITLE
85 add coverage in ci with gcovr

### PR DESCRIPTION
**Pull request by rprat-pro**
Not ready, just to trigger the CI.

### CODE REVIEW

**Comment by rprat-pro** [comment]
@maxence-wz Ready to merge (merge after MR #83 )

**Comment by rprat-pro** [comment]
Conflicts with the new version of the master have been resolved.

File: .github/workflows/cmake.yml [hunk 1]
```diff
@@ -21,6 +21,14 @@ jobs:
         color: true       # Force color output (SPACK_COLOR=always)
         path: ${{github.workspace}}/spack       # Where to clone Spack
 
+    - name: Generate a code coverage repor part1
```
Comment (maxence-wz) [inline .github/workflows/cmake.yml:24]:
```suggestion
    - name: Generate a code coverage report part1
```

File: .github/workflows/cmake.yml [hunk 2]
```diff
@@ -52,7 +60,41 @@ jobs:
...
+      shell: spack-sh {0}    
+      working-directory: ${{github.workspace}}/coverage
       run: | 
         spack load mfem
         spack load tfel
         make check 
+
+    - name: Generate a code coverage report part2
+      working-directory: ${{github.workspace}}/
+      run: gcovr --exclude='build/*' --exclude='tests/*'  --exclude='spack/*'  --exclude='coverage/*' --html-details coverage.html
```
Comment (maxence-wz) [inline .github/workflows/cmake.yml:94]:
```suggestion
      run: |
        gcovr --exclude='build/*'     \
              --exclude='tests/*'     \
              --exclude='spack/*'     \
              --exclude='coverage/*'  \
              --html-details          \
              coverage.html
```

File: .github/workflows/cmake.yml [hunk 3]
```diff
@@ -52,7 +60,41 @@ jobs:
...
+      run: |
+        spack load mfem 
+        cmake -B ${{github.workspace}}/coverage \
+               -DCMAKE_CXX_FLAGS_COVERAGE="--coverage -g" \
+               -DCMAKE_BUILD_TYPE=Coverage -DCMAKE_C_COMPILER=gcc \
+               -DCMAKE_CXX_COMPILER=g++ \
+               -DMFEM_DIR=`spack location -i mfem` \
+               -DHYPRE_DIR=`spack location -i hypre` \
+               -DMFrontGenericInterface_DIR=`spack location -i mgis`/share/mgis/cmake \
+               -DCMAKE_PREFIX_PATH=`spack location -i tfel`
```
Comment (maxence-wz) [inline .github/workflows/cmake.yml:79]:
```suggestion
        cmake -B ${{github.workspace}}/coverage \
              -DCMAKE_CXX_FLAGS_COVERAGE="--coverage -g" \
              -DCMAKE_BUILD_TYPE=Coverage -DCMAKE_C_COMPILER=gcc \
              -DCMAKE_CXX_COMPILER=g++ \
              -DMFEM_DIR=`spack location -i mfem` \
              -DHYPRE_DIR=`spack location -i hypre` \
              -DMFrontGenericInterface_DIR=`spack location -i mgis`/share/mgis/cmake \
              -DCMAKE_PREFIX_PATH=`spack location -i tfel`        
```

**Pull request merged**